### PR TITLE
React UI: Display small numbers correctly

### LINKED ISSUE #6272: React UI: Regression over legend and small values
## Proposal
**Use case. Why is this important?**

This is a regression compared to the current UI

*“Nice to have” is not a good use case. :)*

## Bug Report
**What did you do?**

Run go_gc_duration_seconds

**What did you expect to see?**

legend-old

**What did you see instead? Under which circumstances?**

legend-new

Legend says 0.00, 0.00 , 0.00 ..  instead of 0.003, 0.004, which is strange because it also displays 1.00m, so I would expect to see 3.00m...

**Environment**

* System information:

	linux

* Prometheus version:

prometheus, version 2.13.1 (branch: master, revision: e94503ff5c412590ce7616accdd3c62a2189bcd3)
  build user:       roidelapluie@oxygen
  build date:       20191105-10:36:21
  go version:       go1.13.1

* node version:
v13.0.1

### PR DESCRIPTION
**Pull request by roidelapluie**
Closes #6272

nl

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by hooten** [comment]
Could be cool if there were a unit test for this function

**Comment by roidelapluie** [comment]
Indeed! There is more to fix! wait for it :)

**Comment by roidelapluie** [comment]
@juliusv @hooten this is ready for review.

**Comment by roidelapluie** [comment]
Screnshots
before:
o1

after:
o2

File: web/ui/react-app/src/Graph.test.tsx [hunk 1]
```diff
@@ -101,5 +101,60 @@ describe('Graph', () => {
...
+      expect(graph.formatValue(2e-18)).toBe('2.00a');
+      expect(graph.formatValue(2e-19)).toBe('0.20a');
+      expect(graph.formatValue(2e-20)).toBe('0.02a');
+      expect(graph.formatValue(1e-21)).toBe('1.00z');
+      expect(graph.formatValue(2e-21)).toBe('2.00z');
+      expect(graph.formatValue(2e-22)).toBe('0.20z');
+      expect(graph.formatValue(2e-23)).toBe('0.02z');
+      expect(graph.formatValue(2e-24)).toBe('2.00y');
+      expect(graph.formatValue(2e-25)).toBe('0.20y');
+      expect(graph.formatValue(2e-26)).toBe('0.02y');
```
Comment (juliusv) [inline web/ui/react-app/src/Graph.test.tsx]: How about putting the key/value pairs into an object and then mapping over them?

Comment (roidelapluie) [reply web/ui/react-app/src/Graph.test.tsx]: Indeed, I have done the change.

File: web/ui/react-app/src/Graph.test.tsx [hunk 2]
```diff
@@ -101,5 +101,64 @@ describe('Graph', () => {
...
+        { input: 2e-20, output: '0.02a' },
+        { input: 1e-21, output: '1.00z' },
+        { input: 2e-21, output: '2.00z' },
+        { input: 2e-22, output: '0.20z' },
+        { input: 2e-23, output: '0.02z' },
+        { input: 2e-24, output: '2.00y' },
+        { input: 2e-25, output: '0.20y' },
+        { input: 2e-26, output: '0.02y' },
+      ].map(function(t) {
+        expect(graph.formatValue(t.input)).toBe(t.output);
```
Comment (juliusv) [inline web/ui/react-app/src/Graph.test.tsx]: Heh, I thought about just using the input as the key and the output as the value, but explicit `input` / `output` properties work too :)

**Comment by juliusv** [comment]
:+1: thanks!

File: web/ui/react-app/src/Graph.test.tsx (hunk 2 above)
Comment (roidelapluie) [reply web/ui/react-app/src/Graph.test.tsx]: As in input is not string I made keys.

**Pull request merged**